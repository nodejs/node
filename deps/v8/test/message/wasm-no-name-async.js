// Copyright 2018 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

d8.file.execute('test/mjsunit/mjsunit.js');
d8.file.execute('test/mjsunit/wasm/wasm-module-builder.js');

var builder = new WasmModuleBuilder();
builder.addFunction(undefined, kSig_i_v)
    .addBody([kExprUnreachable])
    .exportAs('main');
let buffer = builder.toBuffer();
assertPromiseResult(WebAssembly.instantiate(buffer), pair => {
  try {
    pair.instance.exports.main();
  } catch (e) {
    print(e.stack);
  }
});
