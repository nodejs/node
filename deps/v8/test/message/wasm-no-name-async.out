RuntimeError: unreachable
    at wasm://wasm/44ee3bce:wasm-function[0]:0x22
    at *%(basename)s:{NUMBER}:27
    at test/mjsunit/mjsunit.js:*
